system/service/queue.go

```diff
@@ -0,0 +1,252 @@
+package service
+
+import (
+	"context"
+
+	"github.com/cortezaproject/corteza-server/pkg/actionlog"
+	"github.com/cortezaproject/corteza-server/pkg/messagebus"
+	"github.com/cortezaproject/corteza-server/store"
+)
+
+type (
+	queue struct {
+		actionlog actionlog.Recorder
+		store     store.Storer
+		ac        queueAccessController
+	}
+
+	QueueService interface {
+		FindByID(ctx context.Context, ID uint64) (*messagebus.QueueSettings, error)
+		Search(context.Context, messagebus.QueueSettingsFilter) (messagebus.QueueSettingsSet, messagebus.QueueSettingsFilter, error)
+
+		Create(ctx context.Context, q *messagebus.QueueSettings) (*messagebus.QueueSettings, error)
+		Update(ctx context.Context, q *messagebus.QueueSettings) (*messagebus.QueueSettings, error)
+
+		DeleteByID(ctx context.Context, ID uint64) error
+		UndeleteByID(ctx context.Context, ID uint64) error
+	}
+
+	queueAccessController interface {
+		CanCreateMessagebusQueue(ctx context.Context) bool
+		CanReadMessagebusQueue(ctx context.Context, c *messagebus.QueueSettings) bool
+		CanUpdateMessagebusQueue(ctx context.Context, c *messagebus.QueueSettings) bool
+		CanDeleteMessagebusQueue(ctx context.Context, c *messagebus.QueueSettings) bool
+
+		CanReadFromMessagebusQueue(ctx context.Context, c *messagebus.QueueSettings) bool
+		CanWriteToMessagebusQueue(ctx context.Context, c *messagebus.QueueSettings) bool
+	}
+)
+
+func Queue() QueueService {
```

Comment:
Please model this service more closely to how Application service is designed.
So, constructor (Queue fn.) should return a ptr to a struct and not an interface.

`...Service` interface inside service pkg (like the `QueueService` above) is an old pattern we need to get rid off across all codebase.